mlfinlab/tests/test_sampling.py

```diff
@@ -86,21 +86,25 @@ def test_seq_bootstrap(self):
         """
 
         non_nan_meta_labels = self.meta_labeled_events.dropna()
-        ind_mat = get_ind_matrix(non_nan_meta_labels)
+        ind_mat = get_ind_matrix(non_nan_meta_labels, self.data)
 
         label_endtime = non_nan_meta_labels.t1
-        bar_index = list(non_nan_meta_labels.index)  # generate index for indicator matrix from t1 and index
+        trimmed_price_bars_index = self.data[(self.data.index >= non_nan_meta_labels.index.min()) &
+                                             (self.data.index <= non_nan_meta_labels.t1.max())].index
+        bar_index = list(non_nan_meta_labels.index)  # Generate index for indicator matrix from t1 and index
         bar_index.extend(non_nan_meta_labels.t1)
-        bar_index = sorted(list(set(bar_index)))  # drop duplicates and sort
+        bar_index.extend(trimmed_price_bars_index)
+        bar_index = sorted(list(set(bar_index)))  # Drop duplicates and sort
         ind_mat_book_implementation = book_ind_mat_implementation(bar_index, label_endtime)
 
         self.assertTrue(bool((ind_mat_book_implementation.values == ind_mat).all()) is True)
-        self.assertTrue(ind_mat.shape == (13, 7))  # Indicator matrix shape should be (meta_label_index+t1, t1)
+        self.assertTrue(
+            ind_mat.shape == (782, 7))  # Indicator matrix shape should be (unique(meta_label_index+t1+bars_index), t1)
         # Check indicator matrix values for specific labels
-        self.assertTrue(bool((ind_mat[:, 0] == [1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]).all()) is True)
-        self.assertTrue(bool((ind_mat[:, 2] == [0, 0, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0]).all()) is True)
-        self.assertTrue(bool((ind_mat[:, 4] == [0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0]).all()) is True)
-        self.assertTrue(bool((ind_mat[:, 6] == [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1]).all()) is True)
+        self.assertTrue(bool((ind_mat[:100, 0] == np.ones(100)).all()) is True)
```

Comment:
This test only checks for 1s but it doesn't check that all the values must not be one. Can we add an assertion to check for some zero elements or to assert that not all are 1s.